### PR TITLE
Add script for multiple output plots

### PR DESCRIPTION
**Pull request by chrisduerr**
This adds the new `gnuplot_separate.sh` script which will do the same
thing as the `gnuplot.sh` script but output each benchmark to a
different file.

This should make it a bit easier to compare two different terminals if
their differences are not significant.

### CODE REVIEW

**Comment by chrisduerr** [comment]
I created this script because I wanted to compare two Alacritty instances directly against each other, without having to deal with multiple combined benchmarks making things difficult to distinguish because of scaling differences.

Do you think this should maybe replace the gnuplot.sh script @kchibisov, or is there value in having both? One might argue that combining everything in a single plot isn't very helpful due to how cluttered things become.

**Review by kchibisov — CHANGES_REQUESTED** [review]
>Do you think this should maybe replace the gnuplot.sh script @kchibisov, or is there value in having both? One might argue that combining everything in a single plot isn't very helpful due to how cluttered things become.

I agree and I'm fine with removing/replacing current `gnuplot.sh`, since it's not really useful and you want only some output to be shown.

I have a question wrt output of a new script, I think it'll make sense to output files into directory and suffix output with `.svg`, so it'll simplify things a bit, and it's likely what you generally want to do, specifying. The name for files in a dir should be just names of the appropriate benchmarks. Since right now it really seems like you want to create a dir(only if it's not presented though) due to how output looks.

**Comment by kchibisov** [comment]
Also, since you've moved scrips into `gnuplot` folder, I think we could remove `gnuplot` prefix from them, since they are already in `gnuplot` folder, what do you think?

**Comment by chrisduerr** [comment]
> I have a question wrt output of a new script, I think it'll make sense to output files into directory and suffix output with .svg, so it'll simplify things a bit

I've thought about a way to simplify it, since the .svg logic was way too complicated. I'm afraid I missed this simple solution completely. I think it's a great idea.

> Also, since you've moved scrips into gnuplot folder, I think we could remove gnuplot prefix from them, since they are already in gnuplot folder, what do you think?

I'm not sure about that, since they should still be self-descriptive as executables? Though `summary.sh` and `detailed.sh` might still be fine.

**Comment by kchibisov** [comment]
>I'm not sure about that, since they should still be self-descriptive as executables? Though summary.sh and detailed.sh might still be fine.

summary.sh and detailed.sh seems fine to me.

File: gnuplot/detailed.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    echo "Usage: gnuplot.sh <INPUT_FILES>... <OUTPUT_DIRECTORY>"
+    exit 1
+fi
+
+# Get last argument as output directory.
+output_index=$#
+output_directory=${!output_index%/}
+mkdir -p "$output_directory"
+
+num_cols=$(cat "$1" | head -n 1 | awk '{ print NF }')
```
Comment (kchibisov) [inline gnuplot/detailed.sh]: I guess you can use `wc -w` instead of `awk`?

File: gnuplot/detailed.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+for col in $(seq 1 $num_cols); do
+    # Append benchmark name before file suffix.
+    benchmark=$(cat "$1" | head -n 1 | awk "{ print \$$col }")
```
Comment (kchibisov) [inline gnuplot/detailed.sh]: I think you can use `cut -d ' ' -f $col` instead of awk?

**Review by kchibisov — CHANGES_REQUESTED** [review]

File: gnuplot/detailed.sh (hunk 2 above)
Comment (chrisduerr) [reply gnuplot/detailed.sh]: But why? That seems significantly more complicated.

File: gnuplot/detailed.sh (hunk 1 above)
Comment (chrisduerr) [reply gnuplot/detailed.sh]: Sure.

File: gnuplot/detailed.sh (hunk 2 above)
Comment (kchibisov) [reply gnuplot/detailed.sh]: it's not, `cut` is a right tool for such things.

Like to understand how `awk` works you should know its syntax, which for example I don't, but `cut` is a default util and its man page is pretty simple

**Review by kchibisov — APPROVED** [review]

**Pull request merged**